runners/spark/src/main/java/org/apache/beam/runners/spark/translation/EvaluationContext.java

```diff
@@ -156,7 +157,7 @@ public Dataset borrowDataset(PValue pvalue) {
     leaves.remove(dataset);
     if (multiReads.contains(pvalue)) {
       // Ensure the RDD is marked as cached
-      dataset.cache();
+      dataset.cache(runtime.getPipelineOptions().as(SparkPipelineOptions.class).getStorageLevel());
```

Comment:
Add a helper method `batchStorageLevel()` ? will come handy in line 178 as well.